src/commands/dev/lib/dev-server.ts

```diff
@@ -90,17 +99,35 @@ export default class DevServer {
     return null;
   }
 
+  async getLocalEnv(fileName: string): Promise<EnvConfig> {
+    const filePath = join(this.cwd, fileName);
+    this.output.debug(`Reading local env: ${filePath}`);
+    let env: EnvConfig = {};
+    try {
+      const dotenv = await fs.readFile(filePath, 'utf8');
+      env = parseDotenv(dotenv);
+    } catch (err) {
+      if (err.code !== 'ENOENT') {
+        throw err;
+      }
+    }
+    return env;
+  }
+
   validateNowConfig(config: NowConfig): void {
     if (config.version !== 2) {
       throw new Error('Only `version: 2` is supported by `now dev`');
     }
-    const buildConfig = config.build || {};
-    const hasSecretEnv = [
-      ...Object.values(config.env || {}),
-      ...Object.values(buildConfig.env || {})
-    ].some(val => val[0] === '@');
-    if (hasSecretEnv) {
-      throw new Error('Secret env vars are not yet supported by `now dev`');
+    this.validateEnvConfig('.env', config.env, this.env);
+    this.validateEnvConfig('.env.build', config.build && config.build.env, this.buildEnv);
+  }
+
+  validateEnvConfig(type: string, env: EnvConfig = {}, localEnv: EnvConfig = {}): void {
+    const missing: string[] = Object.entries(env)
+      .filter(([name, value]) => value.startsWith('@') && !(name in localEnv))
```

Comment:
```suggestion
      .filter(([name, value]) => value.startsWith('@') && !(localEnv.hasOwnProperty(name)))
```

Reply:
`in` is safer to me because of the off-chance where you have a `hasOwnProperty` env var (which granted will probably never be the case, but `in` is more defensive-programming style to me).

Reply:
Then maybe `Object.hasOwnProperty.call(localEnv, name)`? While I'm OK with all these approach, they do probably never be the case :P

Reply:
Ok, updated.